app/services/submission_service.rb

```diff
@@ -104,6 +103,14 @@ def update_submission(
       submission.save!
     end
 
+    def create_or_update_my_collection_artwork(submission, access_token)
+      if !submission.my_collection_artwork_id
+        create_my_collection_artwork(submission, access_token)
+      elsif submission.source == "my_collection"
+        update_my_collection_artwork(submission, access_token)
+      end
```

Comment:
I guess this would set the submission ID on every state change. Can we avoid this and just update the artwork once?